Mathlib/Algebra/Group/Prod.lean

```diff
@@ -447,43 +447,22 @@ variable {M N P : Type*} [Mul M] [Mul N] [Semigroup P]
 /-- Coproduct of two `MulHom`s with the same codomain with commutation assumption :
   `f.noncommCoprod g _ (p : M × N) = f p.1 * g p.2`.
   (For the commutative case, use `MulHom.coprod`) -/
-@[to_additive
+@[to_additive (attr := simps)
     "Coproduct of two `AddHom`s with the same codomain with commutation assumption:
     `f.noncommCoprod g _ (p : M × N) = f p.1 + g p.2`.
     (For the commutative case, use `AddHom.coprod`)"]
 def noncommCoprod : M × N →ₙ* P where
-  toFun mn := (f mn.fst) * (g mn.snd)
-  map_mul' mn mn' := by
-    simp only [Prod.fst_mul, Prod.snd_mul, map_mul]
-    simp only [mul_assoc]
-    apply congr_arg₂ _ rfl
-    simp only [← mul_assoc]
-    apply congr_arg₂ _ _ rfl
-    apply comm
-
-@[to_additive (attr := simp)]
-theorem noncommCoprod_apply (p : M × N) :
-  f.noncommCoprod g comm p = f p.1 * g p.2 :=
-  rfl
-
-/- -- simp_normal_form complains
-@[to_additive (attr := simp)]
-theorem noncommCoprod_apply' (p : M × N) :
-    f.noncommCoprod g comm p = g p.2 * f p.1 :=
-  comm p.1 p.2
--/
+  toFun mn := f mn.fst * g mn.snd
+  map_mul' mn mn' := by simpa using (comm _ _).mul_mul_mul_comm _ _
 
```

Comment:
We shouldn't be committing code in comments; though perhaps this was just you asking for feedback on why `simpNF` complains.

I think this lemma is probably fine to have without the `simp` attribute. You might also want to add the one that says how `noncommCoprod` composes with whatever the name of the multiplicative map `(x, y) => (y, x)` is.

Reply:
I think I can remove it.
What you suggest would be obtained using MulHom.comp, so I'm not sure it is necessary to add it.
(And I don't use noncommProd anymore now…)